das/daser.go

```diff
@@ -89,6 +89,10 @@ func (d *DASer) Start(ctx context.Context) error {
 		return err
 	}
 
+	if err = d.sampler.initMetrics(); err != nil {
```

Comment:
Initializing the metrics shouldn't be done in `Start` if possible - we want to be able to pass a function to the DI when the metrics flag is enabled, so something like `MontitorDASer` in `metrics.go`  would work. Although I see that the challenge here is that getting the stats from the coordinator is private

Reply:
I'm not sure how it is possible to do, considering there are `sync` metrics that are tighten to DASer code and can't be easily disabled. To disable metrics we can just use  dummy metrics collector.
And I really like metrics being `internal` rather than `external` part of DASer. It is so much easier to observe this way.

Reply:
I think @Wondertan suggestion below to do `m == nil` check is the cleanest one and will implement it

Reply:
@Wondertan how will this look in DI?

Reply:
This is the downside of synchronous metrics, and we cannot decouple those from the code. In the DASer case they are kinda justified and the simplest way to solve this would be through the DASer config/options that, besides configuring concurrency params and etc., would also tell DASer whether metrics should be enabled or not.

Reply:
can't we export an `InitMetrics` that just calls this? And then in the DI we can do fx.Invoke that receives the daser? Like

```go
func (d *DASer) InitMetrics() error {
    if err = d.sampler.initMetrics(); err != nil {
        return fmt.Errorf("das: init sampling coordinator metrics: %w", err)
    }
    return nil
}
```

and then


```go
fx.Invoke(func(d *DASer) err {
    return d.InitMetrics()
})
```

Reply:
@distractedm1nd, thanks. This approach is good and is also aligned with our `WithMetrics` option for the node. Not sure rn what is better here

Reply:
@distractedm1nd Thank you! I really like this idea and incorporated this approach in new commit.

Reply:
Looks good now